### ISSUE TITLE
How to use DiceLoss for a single-channel, multiclass target mask (B, W, H) and multi-channel output mask (B, C, W, H)?

### ISSUE THREAD

**Issue by sarmientoj24** [qubvel/segmentation_models.pytorch#542]
I saw in your documentation that you have this DiceLoss.

```
class segmentation_models_pytorch.losses.DiceLoss(mode, classes=None, log_loss=False, from_logits=True, smooth=0.0, ignore_index=None, eps=1e-07)
```

How do i use this for single-channel multiclass target mask and multi-channel output mask?

**Comment by psc0606** [comment]
same question

**Comment by qubvel** [comment]
Hi, thanks for the question, it is expected that you pass single-channel multiclass target masks and multichannel predicted masks, see docs
https://smp.readthedocs.io/en/latest/losses.html#segmentation_models_pytorch.losses.constants.MULTICLASS_MODE
